include/nlohmann/detail/string_concat.hpp

```diff
@@ -0,0 +1,139 @@
+#pragma once
+
+#include <cstring> // strlen
+#include <string> // string
+#include <utility> // forward
+
+#include <nlohmann/detail/meta/cpp_future.hpp>
+#include <nlohmann/detail/meta/detected.hpp>
+
+namespace nlohmann
+{
+namespace detail
+{
+
+inline std::size_t concat_length()
+{
+    return 0;
+}
+
+template<typename... Args>
+inline std::size_t concat_length(const char* cstr, Args&& ... rest);
```

Comment:
Sometimes I think that supporting the requirement that there be a prototype before the declaration is more trouble than it's worth.

Reply:
I've lost quite some time forgetting the prototypes while developing this code and not understanding what the compiler is trying to tell me with seemingly unrelated error messages...